<?xml version="1.0"?>
<launch>
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="verbose" default="false"/>
    <arg name="paused" default="false"/>
    <arg name="respawn_gazebo" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
    	<arg name="world_name" value="$(find aerial_guided_navigation)/worlds/turtlebot3_world.world"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="respawn_gazebo" value="$(arg respawn_gazebo)"/>
    </include> 

    <!-- ROBOT 1 -->
    <group ns="robot1" >
        <param name="tf_prefix" value="robot1_tf" />
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find aerial_guided_navigation)/models/turtlebot/turtlebot3_burger.urdf.xacro"/>
        <include file="$(find aerial_guided_navigation)/launch/turtlebot.launch">
            <arg name="init_pose" value="-x 0.5 -y 0.5 -z 0" />
            <arg name="robot_name" value="turtlebot3"/>
        </include>
    </group>

    <!-- ROBOT 2 -->
    <group ns="robot2"> 
        <param name="tf_prefix" value="robot2_tf"/>
        <param name="robot_description" command="$(find xacro)/xacro --inorder $(find aerial_guided_navigation)/models/iris_downward_camera/iris_downward_camera.urdf"/>
        <include file="$(find aerial_guided_navigation)/launch/drone.launch">
            <arg name="robot_name" value="iris_downward_camera"/>
            <arg name="init_pose" value="-x -0.5 -y 0.5 -z 0 -R 0 -P 0 -Y 0 " />
        </include>
    </group> 
    
    <include file="$(find full_depthimage_to_laserscan)/launch/full_depthimg_to_laser.launch"/>
    
    <!--node pkg="tf" type="static_transform_publisher" name="world_to_robot2_base_link_broadcaster" args="0 0 0 0 0 0 world robot2_tf/odom 100"/-->
    <!--node pkg="tf" type="static_transform_publisher" name="world_to_robot1_odom_broadcaster" args="0 0 0 0 0 0 world robot1_tf/odom 100"/-->
    <node pkg="aerial_guided_navigation" type="odom_publish.py" name="drone_odom"/>
    <node pkg="rviz" type="rviz" name="rviz"/>
</launch>
